run_cert.py

```diff
@@ -0,0 +1,333 @@
+ENV_LETSENCRYPT_SERVER_URL = "LETSENCRYPT_SERVER_URL"
+ENV_LETSENCRYPT_EMAIL = "LETSENCRYPT_EMAIL"
+ENV_VERIFICATION_METHOD = "LETSENCRYPT_VERIFICATION_METHOD"
+ENV_MARATHON_LB_CERT = os.environ.get('MARATHON_LB_CERT_ENV', 'HAPROXY_SSL_CERT')
+ENV_MARATHON_LB_ID = "MARATHON_LB_ID"
+ENV_DOMAINS = "DOMAINS"
+ENV_DNS_PROVIDER = "DNS_PROVIDER"
+DEFAULT_LETSENCRYPT_URL = "https://acme-staging-v02.api.letsencrypt.org/directory"
+ENV_RSA_KEY_SIZE = "RSA_KEY_SIZE"
+DEFAULT_GCE_CREDENTIALS = "/certbot/google_service_account.json"
+ENV_GOOGLE_CREDENTIALS = "GOOGLE_APPLICATION_CREDENTIALS"
+ENV_DNS_PROPAGATION_TIMEOUT = "DNS_PROPAGATION_TIMEOUT"
+ENV_GCE_SERVICE_ACCOUNT = "GCE_SERVICE_ACCOUNT"
+
+CERTIFICATES_DIR = "/etc/letsencrypt/live"
+DOMAINS_FILE = "/etc/letsencrypt/current_domains"
+
+DEFAULT_CERTBOT_ARGS = [
+    "certbot",
+    "certonly",
+    "--server", os.environ.get(ENV_LETSENCRYPT_SERVER_URL, DEFAULT_LETSENCRYPT_URL),
+    "--email", os.environ.get(ENV_LETSENCRYPT_EMAIL),
+    "--agree-tos",
+    "--noninteractive",
+    "--rsa-key-size", os.environ.get(ENV_RSA_KEY_SIZE, "4096"),
+    "--expand"
+]
+
+CERTBOT_ARGS_HTTP = [
+    "--standalone",
+    "--no-redirect",
+    "--preferred-challenges",
+    "http-01"
+]
+
+CERTBOT_ARGS_DNS = [
+    "--preferred-challenges",
+    "dns-01"
+]
+
+CERTBOT_ARGS_DNS_GCLOUD = [
+    "--dns-google",
+    "--dns-google-credentials", os.environ.get(ENV_GOOGLE_CREDENTIALS, DEFAULT_GCE_CREDENTIALS),
+    "--dns-google-propagation-seconds", os.environ.get(ENV_DNS_PROPAGATION_TIMEOUT, "120")
+]
+
+CERTBOT_ARGS_DNS_ROUTE53 = [
+    "--dns-route53",
+    "--dns-route53-propagation-seconds", os.environ.get(ENV_DNS_PROPAGATION_TIMEOUT, "120")
+]
+
+
+def get_marathon_url():
+    """Retrieves the marathon base url to use from an environment variable"""
+    return os.environ.get(ENV_MARATHON_URL, DEFAULT_MARATHON_URL)
+
+
+def get_letsencrypt_url():
+    """Retrieves the LetsEncrypt Server URL"""
+    return os.environ.get(ENV_LETSENCRYPT_SERVER_URL, DEFAULT_LETSENCRYPT_URL)
+
+
+def get_marathon_app(app_id):
+    """Retrieve app definition for marathon-lb app"""
+    response = requests.get("%(marathon_url)s/v2/apps/%(app_id)s"
+                            % dict(marathon_url=get_marathon_url(), app_id=app_id), verify=False)
+    if not response.ok:
+        raise Exception("Could not get app details from marathon")
+    return response.json()
+
+
+def read_domains_from_last_time():
+    """Return list of domains used (last time this script was run) from file or empty string if file does not exist"""
+    if os.path.exists(DOMAINS_FILE):
+        with open(DOMAINS_FILE) as domains_file:
+            return domains_file.read()
+    else:
+        return ""
+
+
+def write_domains_to_file(domains):
+    """Store list of domains in file to retrieve on next run"""
+    with open(DOMAINS_FILE, "w+") as domains_file:
+        domains_file.write(domains)
+
+
+def find_newest_dir(domain_name):
```

Comment:
might be a little simpler with code using max() like this:
https://stackoverflow.com/questions/39327032/how-to-get-the-latest-file-in-a-folder-using-python